source/elements/oneCCL/source/spec/collective_operations.rst

```diff
@@ -39,6 +40,66 @@ The communication operation may accept attribute object. If that parameter is mi
 
 If the arguments provided to a communication operation call do not comply to the requirements of the operation, the behavior is undefined unless it is specified otherwise.
 
+
+.. _Allgather:
+
+Allgather
+*********
+
+Allgather is a collective communication operation that collects the ``send_count`` elements from all the ranks within the communicator and places the results into ``recv_buf``, in such a way that data from rank ``i`` can be found at offset rank ``i * count``. The resulting data in the output ``recv_buf`` buffer is the same for each rank. 
+
+
+Allgather is in place when ``sendbuff == recvbuff + rank * send_count``. 
+
+.. code:: cpp
+
+     template<class BufferType> 
+     event ccl::allgather(const BufferType* send_buf, 
+                          BufferType* recv_buf, 
+                          size_t send_count, 
+                          const communicator& comm, 
+                          const stream& stream, 
+                          const allgather_attr& attr = default_allgather_attr, 
+                          const vector_class<event>& deps = {}); 
+
+     event ccl::allgather(const void* send_buf, 
+                          void* recv_buf, 
+                          size_t send_count, 
+                          datatype dtype,  
+                          const communicator& comm, 
+                          const stream& stream, 
+                          const allgather_attr& attr = default_allgather_attr, 
+                          const vector_class<event>& deps = {}); 
+
+
+
+send_buf 
+    The buffer with send_count elements of BufferType that stores local data to be gathered 
+
+recv_buf [out] 
+    The buffer to store gathered result of BufferTuype, must be large enough to hold values from all ranks, i.e., size should be equal do BufferType * send_count 
+
+send_count 
+    The number of elements of type BufferType in send_buf 
```

Comment:
Here and a few other places you used unicode (non breaking space). You cannot use unicode directly.